core/arch/x86_64/kernel/thread_optee_smc.c

```diff
@@ -301,16 +301,16 @@ static struct mobj *thread_rpc_alloc_arg(size_t size)
 	paddr_t pa;
 	uint64_t co;
 	uint32_t rpc_args[THREAD_RPC_NUM_ARGS] = {
-		OPTEE_SMC_RETURN_RPC_ALLOC, size
+		OPTEE_SMC_RETURN_RPC_ALLOC, (uint32_t)(size & 0x00000000FFFFFFFF)
```

Comment:
This is uint32_t array. Do we need this type casting?

Reply:
Yes, otherwise, the data is not correct.

Reply:
Do you know why there is no this issue on Arm or OP-TEE 3.4 x86?

Reply:
Currently, don't know, we make it work firstly. 
Will do some research later, maybe related to compile process.